src/Mvc/Mvc.Abstractions/ref/Microsoft.AspNetCore.Mvc.Abstractions.netcoreapp3.0.cs

```diff
@@ -209,6 +209,20 @@ public partial interface IAllowAnonymousFilter : Microsoft.AspNetCore.Mvc.Filter
     {
     }
 }
+namespace Microsoft.AspNetCore.Mvc.Diagnostics
+{
+    public abstract partial class MvcDiagnostic : System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.Generic.IReadOnlyCollection<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.Generic.IReadOnlyList<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.IEnumerable
```

Comment:
This should live in `Mvc.Core` there isn't really anything useful to do with the base class by itself. All of the implementation packages reference `Mvc.Core` 

I'd also prefer a name like `EventData` or `DiagnosticEventData` (something without MVC in the name).

Reply:
`EventData` in the `Diagnostics` namespace sounds good